Editor/EditorCore/UVEditor.cs

```diff
@@ -718,44 +718,81 @@ void OnScreenResize()
             Repaint();
         }
 
+        bool IsCopyUVSettingsModifiers(EventModifiers modifiers)
+        {
+            return (modifiers == (EventModifiers.Control | EventModifiers.Shift));
+        }
+
         /**
-         * return true if shortcut should eat the event
+         * returns true if a copy of UV settingsoccured from the first selected face to the target face
          */
-        internal bool ClickShortcutCheck(ProBuilderMesh pb, Face selectedFace)
+        bool CopyFaceUVSettings(ProBuilderMesh pb, Face targetFace)
         {
-            Event e = Event.current;
+            // get first selected UV face
+            ProBuilderMesh firstObj;
+            Face[] source = new Face[1];
 
-            // Copy UV settings
-            if (e.modifiers == (EventModifiers.Control | EventModifiers.Shift))
+            ProBuilderEditor.instance.GetFirstSelectedFace(out firstObj, out source[0]);
+
+            if (source != null)
             {
-                // get first selected Auto UV face
-                ProBuilderMesh firstObj;
-                Face source;
+                UndoUtility.RecordObject(pb, "Copy UV Settings");
 
-                ProBuilderEditor.instance.GetFirstSelectedFace(out firstObj, out source);
+                bool destinationWasManualUV = targetFace.manualUV;
+                bool sourceWasManualUV = source[0].manualUV;
+                if (sourceWasManualUV == true)
+                {
+                    //We need to convert it to auto
+                    UVEditing.SetAutoUV(firstObj, source, true);
+                    pb.Refresh();
+                    pb.Optimize();
```

Comment:
There's no need to call Refresh and Optimize here. They only need to be called when we're done editing the mesh.

Reply:
If I do not call those I get error
![image](https://user-images.githubusercontent.com/48221456/59516777-17f6dd00-8e90-11e9-9d87-e1acaea4d12d.png)

Reply:
All of the ToMesh and Refresh calls can be deferred to the end of the function. I quickly tried it out here, which works: https://github.com/Unity-Technologies/com.unity.probuilder/tree/WB-131_Enable_CopyUV_OfManual_UVs_Edit

Reply:
As mentioned it did not work when the source and target manual/auto did not match. Also when source was manual.